pkg/backup/backup.go

```diff
@@ -167,101 +174,128 @@ func (b *Backup) Run() {
 	}
 
 	if rev <= lastSnapRev {
 		logrus.Info("skipped creating new backup: no change since last time")
-		return lastSnapRev, nil
-	}
-
-	log.Printf("saving backup for cluster (%s)", b.clusterName)
-	if err := b.writeSnap(member, rev); err != nil {
-		err = fmt.Errorf("write snapshot failed: %v", err)
-		return lastSnapRev, err
+		return 0, 0, "", fmt.Errorf("skipped creating new backup: no change since last time")
 	}
-	return rev, nil
-}
-
-func (b *Backup) writeSnap(m *etcdutil.Member, rev int64) error {
-	start := time.Now()
 
 	cfg := clientv3.Config{
 		Endpoints:   []string{m.ClientURL()},
 		DialTimeout: constants.DefaultDialTimeout,
-		TLS:         b.etcdTLSConfig,
+		TLS:         tls,
 	}
+	ctx, cancel := context.WithTimeout(context.Background(), constants.DefaultRequestTimeout)
 	etcdcli, err := clientv3.New(cfg)
+	defer etcdcli.Close()
 	if err != nil {
-		return fmt.Errorf("failed to create etcd client (%v)", err)
+		return 0, 0, "", fmt.Errorf("failed to create etcd client (%v)", err)
 	}
-	defer etcdcli.Close()
 
-	ctx, cancel := context.WithTimeout(context.Background(), constants.DefaultRequestTimeout)
-	resp, err := etcdcli.Maintenance.Status(ctx, m.ClientURL())
+	resp, err := etcdcli.Maintenance.Status(ctx, cfg.Endpoints[0])
 	cancel()
 	if err != nil {
-		return err
+		return 0, 0, "", fmt.Errorf("failed to get member status (%v)", err)
 	}
 
+	etcdVersion := resp.Version
+
 	ctx, cancel = context.WithTimeout(context.Background(), constants.DefaultSnapshotTimeout)
 	rc, err := etcdcli.Maintenance.Snapshot(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to receive snapshot (%v)", err)
+		return 0, 0, "", fmt.Errorf("failed to get member snapshot (%v)", err)
 	}
-	defer cancel()
-	defer rc.Close()
 
-	n, err := b.be.Save(resp.Version, rev, rc)
+	n, err := be.Save(etcdVersion, rev, rc)
+	cancel()
+	rc.Close()
 	if err != nil {
-		return err
+		return 0, 0, "", err
 	}
+	return n, rev, etcdVersion, nil
+}
 
-	bs := backupapi.BackupStatus{
-		CreationTime:     time.Now().Format(time.RFC3339),
-		Size:             util.ToMB(n),
-		Version:          resp.Version,
-		Revision:         rev,
-		TimeTookInSecond: int(time.Since(start).Seconds() + 1),
+func (b *BackupManager) getLatestBackupRev() int64 {
+	// If there is any error, we just exit backup sidecar because we can't serve the backup any way.
+	name, err := b.be.GetLatest()
+	if err != nil {
+		logrus.Fatal(err)
 	}
-	b.recentBackupsStatus = append(b.recentBackupsStatus, bs)
-	if len(b.recentBackupsStatus) > maxRecentBackupStatusCount {
-		b.recentBackupsStatus = b.recentBackupsStatus[1:]
+	if len(name) == 0 {
+		return 0
+	}
+	rev, err := util.GetRev(name)
+	if err != nil {
+		logrus.Fatal(err)
 	}
+	return rev
+}
+
+func (b *BackupManager) getLatestBackupStatus() backupapi.BackupStatus {
+	return b.recentBackupsStatus[len(b.recentBackupsStatus)-1]
+}
 
-	return nil
+func getRunningPods(kclient kubernetes.Interface, namespace, clusterName string) ([]*v1.Pod, error) {
```

Comment:
revert this change? is this relevant to what we want to achieve?

Reply:
not anymore. will revert.